src/miner.cpp

```diff
@@ -624,6 +623,104 @@ bool CreateCoinStake( CBlock &blocknew, CKey &key,
     return false;
 }
 
+
+
+void SplitCoinStakeOutput(CBlock &blocknew)
+{
+    // When this function is called, CreateCoinStake and CreateGridcoinReward have already been called
+    // and there will be a single coinstake output (besides the empty one) that has the combined stake + research
```

Comment:
great candidate for assert()ion

Reply:
What are you suggesting to assert? There is no condition in any of that code that is expected to fail. It is a no-op if the number of UTXO's is 1, and GetNumberofStakeOutputs is clamped to 2. If CreateCoinStake and CreateGridcoinReward have succeeded (because we have arrived at the call to SplitCoinStakeOutput, then CBlock StakeBlock already exists and is in the proper form.

Reply:
assert(blockNew.vtx.size() == 2)

Reply:
Ok

Reply:
@tomasbrod I think you actually meant assert(blockNew.vtx[1].vout.size()==2)...